content/utils/ai-utils.js

```
@@ -0,0 +1,269 @@
+    };
+  }
+
+  /**
+   * Pre-process highlight text for better AI summarization
+   */
+  preprocessHighlightText(text) {
+    // Remove excessive whitespace
+    text = text.replace(/\s+/g, " ").trim();
+
+    // Remove non-printable characters
+    text = text.replace(/[\x00-\x08\x0B\x0C\x0E-\x1F\x7F]/g, "");
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)